chain/bitcoind_client.go

```diff
@@ -442,27 +466,53 @@ func (c *BitcoindClient) RescanBlocks(
 		if !c.shouldFilterBlock(time.Unix(header.Time, 0)) {
 			continue
 		}
+		filteredHeaders[hash] = header
+		blockHashPointers = append(blockHashPointers, &hash)
+	}
+
+	rescannedBlocks := make([]btcjson.RescannedBlock, 0, len(filteredHeaders))
+
+	// Now we fetch blocks in interval of 15 to avoid out of memory errors in
+	// case of fetching too many blocks with GetBlocksBatch().
+	const batchSize = 15
+	total := len(blockHashPointers)
+
+	for i := 0; i < total; i += batchSize {
+		start := i
+		end := i + batchSize
+		if end > total {
+			end = total
+		}
 
-		block, err := c.GetBlock(&hash)
+		blocks, err := c.chainConn.GetBlocksBatch(
+			blockHashPointers[start:end],
+		)
 		if err != nil {
-			log.Warnf("Unable to get block %s from bitcoind: %s",
-				hash, err)
-			continue
+			return nil, err
 		}
+		batchedBlockHashPointers := blockHashPointers[start:end]
+		index := -1
```

Comment:
nit: not sure what the role of the variable `index` is?

Reply:
Yes, this is redundant now since I am traversing using index, it was there as per earlier implementation which used maps for traversal. Will fix it thanks for pointing it out!